### PR TITLE
feat: HogQL shortcuts in taxonomic filters

### LINKED ISSUE #13172: Allow breakdown by `day of week` and `hour of day`
Currently it's not possible to view a frequency bucketed by hour of day or day of week. Both analyses are common and it would be good to support them.

Alternatives: wait for sql editor, do nothing

### PR DESCRIPTION
**Pull request by anirudhpillai**
## Problem
Fixes #13172 

Letting people breakdown by 'day of week', 'hour' without having to know about/type in HogQL.

## Changes
Initial version of UI, wanted quick feedback/open to changes.
image

image

The `InfiniteList` is very tightly integrated within in the `TaxonomicFilter` like [here](https://github.com/PostHog/posthog/blob/feature/hogql-shortcuts/frontend/src/lib/components/TaxonomicFilter/InfiniteSelectResults.tsx#L44) for eg. Can still try to integrate the `render` [component](https://github.com/PostHog/posthog/blob/feature/hogql-shortcuts/frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx#L491) better but wanted to check if it's worth the effort. Integrating will show the user the filtered counts in the category pill and make items searchable. But the search is already good enough in that it falls back to the hogql tab if it doesn't find anything else.
Also I think integrating them closely might be a bit messy with lot of overrides for this specific scenario, though might have missed a cleaner way to do it.

I went with the approach the Replay properties are using
image

## Does this work well for both Cloud and self-hosted?
N/A

## How did you test this code?
Tested on UI

### CODE REVIEW

File: frontend/src/lib/components/TaxonomicFilter/InlineHogQLEditor.tsx [hunk 1]
```diff
@@ -9,18 +10,47 @@ export interface InlineHogQLEditorProps {
     metadataSource?: AnyDataNode
 }
 
+const SHORTCUTS = [
```
Comment (anirudhpillai) [inline frontend/src/lib/components/TaxonomicFilter/InlineHogQLEditor.tsx:13]: Guessing can expose things to use latest event/person property
```
select 
   event, 
   pdi.person.properties.email, -- current property value, fetched via a join
   poe.properties.email, -- ingestion time property stored on the event
   person.properties.email -- whatever is the default setting for your team, one of the two above
from events limit 10
```

**Review by Twixes — COMMENTED** [review]
Sounds good to me at a high level. 

Two things that stand out to me at a glance: 1. there's a lot of unused whitespace due to the column-based layout, 2. the "Enter HogQL Expression, such as" list is conceptually _pretty_ similar to shortcuts – both are lists of examples, except the first one is kind of raw, while the second is ready-made.

The question is: how can we make this user experience the best it can be?

Screenshot 2024-07-19 at 13 10 49

**Comment by anirudhpillai** [comment]
Sorry should have specified earlier.
1. I pictured the column on the right to be scrollable (once we add more shortcuts especially [latest/events](https://github.com/PostHog/posthog/pull/23824#discussion_r1683114653)). 2 column layout made more sense as then you can access shortcuts and both the HogQL editor quickly (over putting the shortcuts under the editor).
2. I decided against hiding the placeholder as the column on the right would need more space once we have more shortcuts. Though was also thinking maybe it made more sense to change the placeholder to be something like the snippet [here](https://github.com/PostHog/posthog/pull/23824#discussion_r1683114653) so users know how to access different property values.

File: frontend/src/lib/components/TaxonomicFilter/InlineHogQLEditor.tsx (hunk 1 above)
Comment (anirudhpillai) [reply frontend/src/lib/components/TaxonomicFilter/InlineHogQLEditor.tsx:13]: Skipping these for now as we'll implement this in the event/person properties picker

**Comment by anirudhpillai** [comment]
Moved to a single column UI
Can use ideas to restyle the 'shortcuts' label. Think we might be able to get away without the label as (which looks better) as we already have tooltips for the shortcuts and when the user clicks them the behaviour is easy to understand - it fills the editor with the tooltip and sets the breakdown.

image

**Review by Twixes — CHANGES_REQUESTED** [review]
I've been thinking about this UI. The current iteration feels alright… but not _great_. I think there's a couple of aspects to it:
1. Having the list of shortcuts on the side creates a bunch of useless whitespace. We must strive for compactness.
2. The shortcuts alongside the examples feel kind of clunky. Specifically, the examples feel very stop-gappy.

I have two ideas on this:
1. The whitespace problem can be resolved by moving shortcuts directly below the expression editor
2. I think we should outsource the current examples to the Docs. Specifically, we can create a "Cheatsheet" page in the existing HogQL section of the docs in posthog.com, and link to it in `HogQLEditor`. That cheatsheet would then neatly open in the docs side panel, so it'd actually be handy. Maybe a little bit inspired by regexr.com's regex cheatsheet – except ours will be SQL:
    Screenshot 2024-08-01 at 13 17 16

**Pull request closed without merging**